flink-runtime/src/main/java/org/apache/flink/runtime/entrypoint/component/DispatcherResourceManagerComponent.java

```diff
@@ -135,6 +135,22 @@ private void registerShutDownFuture() {
         }
     }
 
+    /**
+     * Close the web monitor and cluster components. This method will not deregister the Flink
+     * application from the resource management.
+     *
+     * @return Future which is completed once the shut down
+     */
+    public CompletableFuture<Void> closeAsyncWithoutDeregisteringApplication() {
+
+        if (isRunning.compareAndSet(true, false)) {
+            return FutureUtils.composeAfterwards(
+                    webMonitorEndpoint.closeAsync(), this::closeAsyncInternal);
+        } else {
+            return terminationFuture;
+        }
+    }
```

Comment:
I would suggest to not duplicate the shut down logic. Having a single place where the shut down happens has the benefit that only one place needs to be updated in case of a change. Maybe we can have an internal method which takes an additional shutdown `Runnable` which in case of the deregister application case calls `deregisterApplication()` and in the non deregister application call is a no-op. What do you think?